homeassistant/components/sensor/recorder.py

```diff
@@ -499,10 +499,32 @@ def compile_statistics(  # noqa: C901
 
         result[entity_id]["stat"] = stat
 
+    statistic_ids = statistics.list_statistic_ids(hass, "sum")
+    for statistic_id in statistic_ids:
+        if split_entity_id(statistic_id)[0] != DOMAIN:
+            continue
+        if statistic_id in result:
+            continue
+        last_stats = statistics.get_last_statistics(hass, 1, statistic_id, False)
```

Comment:
Would it make sense to put this logic inside the above loop? Basically set `new_state = old_state` and let all older logic just kick in ?